src/sagemaker_containers/encoders.py

```diff
@@ -72,30 +91,34 @@ def default(_array_like):
     return json.dumps(array_like, default=default)
 
 
-def json_to_numpy(string):  # type: (object) -> np.array
+def json_to_numpy(string_like, dtype=np.float32):  # type: (str or unicode) -> np.array
     """Convert a JSON object to a numpy array.
 
         Args:
-            string (str): JSON string.
+            dtype (dtype, optional):  Data type of the resulting array.
+                                    If None, the dtypes will be determined by the contents of each column, individually.
+            string_like (str): JSON string.
 
         Returns:
             (np.array): numpy array
         """
-    data = json.loads(string)
-    return np.array(data)
+    data = json.loads(unicode_to_str(string_like))
+    return np.array(data).astype(dtype=dtype)
```

Comment:
You can also pass in the dtype to `np.array`.

Reply:
^^ but it works slightly differently. i.e. constructor arg won't downcast, but astype will.